
<style>
  .rtl { direction:rtl;}
  .b {line-height: 200%;}
</style>

 <div  class="b">
 <p style="text-align:right;">
     שבץ (באנגלית: Stroke או Cerebro Vascular Accident ובקיצור CVA) הוא אירוע חירום רפואי, שעלול להביא לפגיעה עצבית קבועה, לסיבוכים ולמוות. הוא הגורם העיקרי לנכות אצל מבוגרים בעולם המערבי והסיבה השנייה למוות ברחבי העולם. בישראל, לפי נתוני משרד הבריאות, מתרחשים כ-15,000 מקרים חדשים של שבץ מוחי מדי שנה. שבץ מתרחש כאשר ישנה הפרעה פתאומית באספקת הדם לרקמת המוח. ההפרעה הזאת גורמת נזק נוירולוגי אשר יכול לבוא לידי ביטוי בדרכים שונות ובהן הפרעה בדיבור, חולשה או שיתוק של הגפיים וקשיים בהליכה. לעיתים עלול אירוע מוחי לגרום למוות. התרחשות של שבץ היא תהליך מהיר שבסופו, בגלל הפגיעה בזרימת הדם למוח, תאי מוח ניזוקים והתפקוד שעליו אחראי האזור הפגוע במוח ניזוק גם הוא. לדוגמה, אם נפגע האזור במוח ששולט על יכולת הדיבור, ייפגעו התפקוד השפתי ומרכז הדיבור של החולה
    </p>
   
    
 </div> 



<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
    <h4>  
את הנתונים כבר ניתחו הרבה מדעני נתונים לכן מה שאציג כאן הוא איך לגשת למידע ולנתח אותו 
 

    
</h4> </div> 

https://www.kaggle.com/fedesoriano/stroke-prediction-dataset/download


 <style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
    נייבא כמה סיפריות בסיסיות שנרחיב עליהן בפוסטים נפרדים 

    
 </div> 
    


In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
כאשר אנחנו מגדירים מחלקה חדשה ב-CSS, עלינו לפתוח סוגריים מסולסלים ולהגדיר את התכונות של כל מחלקה ומחלקה. עבור המחלקה ```rtl``` לדוגמה, הגדרנו שכיוון הכתיבה הוא מימין לשמאל, לכן בחרנו להשתמש בתכונה ```direction``` שמקבלת את הערך ```rtl```.
</div>


<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
נגדיר משתנה src בתור המקור ממנו נייבא את הנתונים ונשתמש בפונקציה בשם read_csv מתוך ספרית Pandas
</div>

In [2]:
src = r'C:\Users\ayose\GDrive\Data Science\Stroke'
s_data = pd.read_csv(src+'/healthcare-dataset-stroke-data.csv')



    
<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
דבר ראשון נסתכל על הנתונים מלמעלה:
</div>

In [3]:
s_data.head(3)

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   

   stroke  
0       1  
1       1  
2       1

In [4]:
s_data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


In [8]:
s_data.drop(columns=['id']).describe()

age  hypertension  heart_disease  avg_glucose_level  \
count  5110.000000   5110.000000    5110.000000        5110.000000   
mean     43.226614      0.097456       0.054012         106.147677   
std      22.612647      0.296607       0.226063          45.283560   
min       0.080000      0.000000       0.000000          55.120000   
25%      25.000000      0.000000       0.000000          77.245000   
50%      45.000000      0.000000       0.000000          91.885000   
75%      61.000000      0.000000       0.000000         114.090000   
max      82.000000      1.000000       1.000000         271.740000   

               bmi       stroke  
count  4909.000000  5110.000000  
mean     28.893237     0.048728  
std       7.854067     0.215320  
min      10.300000     0.000000  
25%      23.500000     0.000000  
50%      28.100000     0.000000  
75%      33.100000     0.000000  
max      97.600000     1.000000

 <div dir="auto">
כאשר אנחנו מגדירים מחלקה חדשה ב-CSS, עלינו לפתוח סוגריים מסולסלים ולהגדיר את התכונות של כל מחלקה ומחלקה. עבור המחלקה ```rtl``` לדוגמה, הגדרנו שכיוון הכתיבה הוא מימין לשמאל, לכן בחרנו להשתמש בתכונה ```direction``` שמקבלת את הערך ```rtl```.
</div>


In [10]:
# ראשית נבדוק כמה נתונים פגומים יש לנו
s_data.isna().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64


<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
ניתן לראות שבערכי ה BMI ישנם 201 ערכים לא מספריים (NaN)

</div>

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

In [9]:
from nb2wp import nb2wp


ModuleNotFoundError: No module named 'nb2wp'

In [1]:
!jupyter nbconvert --to html Stroke.ipynb

'jupyter' is not recognized as an internal or external command,
operable program or batch file.
